src/aggregation/bucket/histogram/histogram.rs

```diff
@@ -0,0 +1,822 @@
+use serde::{Deserialize, Serialize};
+
+use crate::aggregation::agg_req_with_accessor::{
+    AggregationsWithAccessor, BucketAggregationWithAccessor,
+};
+use crate::aggregation::f64_from_fastfield_u64;
+use crate::aggregation::intermediate_agg_result::IntermediateBucketResult;
+use crate::aggregation::segment_agg_result::{
+    SegmentAggregationResultsCollector, SegmentHistogramBucketEntry,
+};
+use crate::fastfield::{DynamicFastFieldReader, FastFieldReader};
+use crate::schema::Type;
+use crate::{DocId, TantivyError};
+
+/// Histogram is a bucket aggregation, where buckets are created dynamically for given `interval`.
+/// Each document value is rounded down to its bucket.
+///
+/// E.g. if we have a price 18 and an interval of 5, the document will fall into the bucket with
+/// the key 15. The formula used for this is:
+/// `((val - offset) / interval).floor() * interval + offset`
+///
+/// For this calculation all fastfield values are converted to f64.
```

Comment:
I wonder if this is a good idea. u64 -> f64 is not super cheap and will yield wrong results sometimes.
On the other hand, the main use for this is timestamp which are i64/u64